beacon-chain/sync/subscriber_beacon_blocks.go

```diff
@@ -55,6 +58,79 @@ func (s *Service) beaconBlockSubscriber(ctx context.Context, msg proto.Message)
 	return err
 }
 
+// reconstructAndBroadcastBlobs processes and broadcasts blob sidecars for a given beacon block.
+// This function reconstructs the blob sidecars from the EL using the block's KZG commitments,
+// broadcasts the reconstructed blobs over P2P, and saves them into the blob storage.
+func (s *Service) reconstructAndBroadcastBlobs(ctx context.Context, block interfaces.ReadOnlySignedBeaconBlock) {
+	startTime, err := slots.ToTime(uint64(s.cfg.chain.GenesisTime().Unix()), block.Block().Slot())
+	if err != nil {
+		log.WithError(err).Error("Failed to convert slot to time")
+	}
+
+	blockRoot, err := block.Block().HashTreeRoot()
+	if err != nil {
+		log.WithError(err).Error("Failed to calculate block root")
+		return
+	}
+
+	if s.cfg.blobStorage == nil {
+		return
+	}
+	indices, err := s.cfg.blobStorage.Indices(blockRoot)
+	if err != nil {
+		log.WithError(err).Error("Failed to retrieve indices for block")
+		return
+	}
+	for _, index := range indices {
+		if index {
+			blobExistedInDBCount.Inc()
+		}
+	}
+
+	// Reconstruct blob sidecars from the EL
+	blobSidecars, err := s.cfg.executionReconstructor.ReconstructBlobSidecars(ctx, block, blockRoot, indices)
+	if err != nil {
+		log.WithError(err).Error("Failed to reconstruct blob sidecars")
+		return
+	}
+	if len(blobSidecars) == 0 {
+		return
+	}
+
+	// Refresh indices as new blobs may have been added to the db
+	indices, err = s.cfg.blobStorage.Indices(blockRoot)
+	if err != nil {
+		log.WithError(err).Error("Failed to retrieve indices for block")
+		return
+	}
+
+	// Broadcast blob sidecars first than save them to the db
+	for _, sidecar := range blobSidecars {
+		if sidecar.Index >= uint64(len(indices)) || indices[sidecar.Index] {
+			continue
+		}
+		if err := s.cfg.p2p.BroadcastBlob(ctx, sidecar.Index, sidecar.BlobSidecar); err != nil {
+			log.WithFields(blobFields(sidecar.ROBlob)).WithError(err).Error("Failed to broadcast blob sidecar")
+		}
+	}
+
+	for _, sidecar := range blobSidecars {
+		if sidecar.Index >= uint64(len(indices)) || indices[sidecar.Index] {
+			blobExistedInDBCount.Inc()
+			continue
+		}
+		if err := s.subscribeBlob(ctx, sidecar); err != nil {
+			log.WithFields(blobFields(sidecar.ROBlob)).WithError(err).Error("Failed to receive blob")
+			continue
+		}
+
+		blobRecoveredFromELCount.Inc()
+		fields := blobFields(sidecar.ROBlob)
+		fields["sinceSlotStartTime"] = s.cfg.clock.Now().Sub(startTime)
+		log.WithFields(fields).Debug("Processed blob sidecar from EL")
```

Comment:
If latency is very important here, consider a log level check before computing the fields

```suggestion
		if logrus.GetLevel() >= logrus.DebugLevel {
			fields := blobFields(sidecar.ROBlob)
			fields["sinceSlotStartTime"] = s.cfg.clock.Now().Sub(startTime)
			log.WithFields(fields).Debug("Processed blob sidecar from EL")
		}
```

Reply:
It's not very important here. It's more important for the previous for loop (broadcast), but these are fast anyway, so we should be ok...